style/testing-jest/README.md

```diff
@@ -0,0 +1,28 @@
+# Testing with Jest
+
+See also [Testing](../testing/README.md).
+
+* Use [eslint-plugin-jest] to enforce testing style
+* Use [testing-library/jest-dom] and [jest-community/jest-extended] for
+  supplemental expectation matchers
+* Use [React Testing Library] for testing [React](../react/README.md)
+  components
+* Use [React Hooks Testing Library] for testing [React Hooks]
+* Use [User Event] for simulating DOM events on React components under test
+* Use [Fishery] for building factories
+* Prefer placing test suite files alongside source files (e.g. `Thing.tsx`
```

Comment:
A pattern I've seen more often is to put tests in `./__tests__/Thing.test.tsx`. I like that pattern since it keeps tests a bit out of the way but also almost colocated.

Reply:
I really like having the test files as siblings of the component they test, though that may be a result of how I'm used to searching for them.

One benefit of the `__tests__` folder is that it does reduce noise in a directory where there are a lot of components. But I have also found it makes it a little harder to, at a glance, see if a component has a test suite.

I'm also open to either way - curious to hear others' thoughts on this!

Reply:
I too prefer the tests as siblings! I find that the specs are better maintained when they arent hidden off in a folder.  It seems to elevate the tests to more of a first class concern, which imo they should be.

Reply:
I am of a third opinion that tests should not be alongside components at all and instead be in their own root level directory that mirrors the components directory. Basically how most Rails projects (along with [many](https://github.com/luckyframework/lucky) [other](https://github.com/lefticus/cpp_starter_project) [languages](https://github.com/rust-lang/rustfmt) and [frameworks](https://github.com/rappasoft/laravel-boilerplate)) treat tests.

Although the entire community seems to be more in favor of tests near components and we'd be against almost every project we're likely to work with. I also see John's point about tests being a first-class citizen and I agree, but I have not found them to be more or less maintained because they are near or far in the file system. That feels like it's been more of a cultural problem than a test placement one.

If these are the options, I would at least prefer they are out of the way a little in a sibling folder. I think it makes them easier to work with which could decrease the friction of maintain them.

Reply:
I have created https://github.com/thoughtbot/guides/issues/613 to continue this discussion